### ISSUE TITLE
Is "create" a reserved word?

### ISSUE THREAD

**Issue by AlexCheema** [xf00f/web3x#24]
Not sure what exactly causes this. I managed to reproduce the bug with a simple contract. It seems that the arguments to the contract are important, as a function with no arguments does not reproduce the bug.

Very simple contract to reproduce:
```solidity
pragma solidity ^0.5.0;

contract ReproduceBug {

    address test;

    constructor(address _test) public {
    }

    function create(uint256 tokenId, address itemOwner, bytes32[] memory keys, bytes32[] memory values) public returns (bool) {
        return true;
    }

}
```

Gets built to:
```typescript
import BN from "bn.js";
import { Address } from "web3x/address";
import { EventLog, TransactionReceipt } from "web3x/formatters";
import { Contract, ContractOptions, TxCall, TxSend, TxDeploy, EventSubscriptionFactory } from "web3x/contract";
import { Eth } from "web3x/eth";
import abi from "./ReproduceBugAbi";
interface ReproduceBugEvents {
}
interface ReproduceBugEventLogs {
}
interface ReproduceBugTxEventLogs {
}
export interface ReproduceBugTransactionReceipt extends TransactionReceipt<ReproduceBugTxEventLogs> {
}
interface ReproduceBugMethods {
    create(tokenId: number | string | BN, itemOwner: Address, keys: string[], values: string[]): TxSend<ReproduceBugTransactionReceipt>;
}
export interface ReproduceBugDefinition {
    methods: ReproduceBugMethods;
    events: ReproduceBugEvents;
    eventLogs: ReproduceBugEventLogs;
}
export class ReproduceBug extends Contract<ReproduceBugDefinition> {
    constructor(eth: Eth, address?: Address, options?: ContractOptions) {
        super(eth, abi, address, options);
    }
    deploy(_test: Address): TxSend<ReproduceBugTransactionReceipt> {
        return super.deployBytecode("0x608060405234801561001057600080fd5b506040516020806102668339810180604052602081101561003057600080fd5b810190808051906020019092919050505050610215806100516000396000f3fe608060405234801561001057600080fd5b5060043610610048576000357c01000000000000000000000000000000000000000000000000000000009004806304d36f081461004d575b600080fd5b6101c16004803603608081101561006357600080fd5b8101908080359060200190929190803573ffffffffffffffffffffffffffffffffffffffff169060200190929190803590602001906401000000008111156100aa57600080fd5b8201836020820111156100bc57600080fd5b803590602001918460208302840111640100000000831117156100de57600080fd5b919080806020026020016040519081016040528093929190818152602001838360200280828437600081840152601f19601f8201169050808301925050505050505091929192908035906020019064010000000081111561013e57600080fd5b82018360208201111561015057600080fd5b8035906020019184602083028401116401000000008311171561017257600080fd5b919080806020026020016040519081016040528093929190818152602001838360200280828437600081840152601f19601f8201169050808301925050505050505091929192905050506101db565b604051808215151515815260200191505060405180910390f35b60006001905094935050505056fea165627a7a72305820ebaafbde68517ea43a3c7558394ababa863d10665bf22c9d05691dbb010a15cc0029", _test) as any;
    }
}
export var ReproduceBugAbi = abi;
```

And a simple test I am using to reproduce it. The full setup works and when I change the name of the "create" function to "createTest" or anything else, it works!

```typescript
import { startLocalBlockchain } from "@cos/blockchain";
import { Address } from "web3x/address";
import { formatBytes32String } from "web3x/ethers/utf8";
import { ReproduceBug } from "../../src/contracts/cos/ReproduceBug";

test("reverts when function name is create", async () => {
  const { eth, blockchain } = await startLocalBlockchain({ hostname: "127.0.0.1", port: 8500, findNextPort: true });

  const from = Address.fromString(Object.keys(blockchain.unlocked_accounts)[0]);

  const buggedContract = new ReproduceBug(eth);
  await buggedContract
    .deploy(from)
    .send({ from, gas: 40000000 })
    .getReceipt();
  console.log("Test contract addr", buggedContract.address);
  const result = await buggedContract.methods
    .create("1", Address.ZERO, [formatBytes32String("hallo")], [formatBytes32String("yoyo")])
    .send({ from, gas: 40000000 })
    .getReceipt();
  console.log("Got test reuslt", result);
});
```

Using truffle v0.5.4 to build the contract and web3x v3.0.1. Did not have this issue on version 2.

**Comment by AlexCheema** [comment]
I just had the same issue. A contract I am using inherits from ERC721. ERC721 has an approve function. Every time I called this, it would instantly get reverted. Making another function called "approveToken" which in turn calls "approve" fixed the issue.

It seems there is an issue in the way you resolve contract functions. Perhaps there is a bug in this code that causes collisions between different functions with similar signatures or something like that?

**Comment by xf00f** [comment]
Apologies for the problems and thanks for reporting. I'll investigate and report back.

**Comment by AlexCheema** [comment]
No problem. Thank you!

**Comment by xf00f** [comment]
@AlexCheema This should be fixed in v3.0.2. There was a padding issue causing the function signature to lose a leading 0 if one was present. Thank you for reporting.
